src/renderer/hooks/useInstantOnboarding.ts

```diff
@@ -0,0 +1,144 @@
+          )
+        }
+      }
+
+      setShowInstantOnboarding(true)
+
+      // Since a QR code can be scanned by the user in literally any situation,
+      // also when the user already has an existing account and is logged in, we
+      // can store it here temporarily, adjust the UI state and bring the user
+      // back to the QR code process again when everything is ready.
+      setWelcomeQr(qrWithUrl)
+    },
+    [setWelcomeQr, setShowInstantOnboarding]
+  )
+
+  const createInstantAccount = useCallback(
+    async (
+      accountId: number,
+      displayName: string,
+      profilePicture?: string
+    ): Promise<T.FullChat['id'] | null> => {
+      let instanceUrl = `dcaccount:${DEFAULT_CHATMAIL_INSTANCE_URL}`
+
+      // Use custom chatmail instance if given by QR code
+      if (welcomeQr && welcomeQr.qr.kind === 'account') {
+        instanceUrl = welcomeQr.url
+      }
+
+      // 1. In this "Instant Onboarding" account creation flow the user is not
+      // asked to manually insert any mail server credentials.
+      //
+      // Internally, the function will call the chatmail instance URL via HTTP
+      // from which it'll receive the address and password as an JSON object.
+      //
+      // After parsing it'll automatically configure the account with the
+      // appropriate keys for login, e.g. `addr` and `mail_pw`
+      await BackendRemote.rpc.setConfigFromQr(accountId, instanceUrl)
+
+      return new Promise((resolve, reject) => {
+        // 2. Kick-off the actual account creation process by calling
+        // `configure`. This happens inside of this dialog
+        openDialog(ConfigureProgressDialog, {
+          onSuccess: async () => {
+            try {
+              // 3. Additionally we set the `selfavatar` / profile picture
+              // and `displayname` configuration for this account.
+              //
+              // Note: We need to set these profile settings _after_
+              // calling `configure` to make the UI aware of them already
+              if (profilePicture) {
+                await BackendRemote.rpc.setConfig(
+                  accountId,
+                  'selfavatar',
+                  profilePicture
+                )
+              }
+              await BackendRemote.rpc.setConfig(
+                accountId,
+                'displayname',
+                displayName
+              )
+
+              // 4. If the user created a new account from trying to contact
+              // another user or joining the group we continue with this now
+              let chatId: number | null = null
+              if (welcomeQr) {
+                if (welcomeQr.qr.kind === 'askVerifyContact') {
+                  chatId = await secureJoinContact(accountId, welcomeQr, true)
+                } else if (welcomeQr.qr.kind === 'askVerifyGroup') {
+                  chatId = await secureJoinGroup(accountId, welcomeQr, true)
+                }
```

Comment:
if we do not accept any qr code here then we should maybe log that we received an unexpected qr type here?
I guess we'll have at least one more case: opening a mailto link with DC before having a configured account.

An error log entry could help the devs in the future to figgure out that they need to add their qr code handling case here, that it works.

Reply:
> if we do not accept any qr code here then we should maybe log that we received an unexpected qr type here?

If I understand your question then this is already taken care of in other places. See `switchToInstantOnboarding` and the `useProcessQR` logic

Reply:
I think it still makes sense to log the unexpected qr code as error or warning to help future devs.

Reply:
Doing it with typescript is an even better idea, thanks. I made a commit to make it fail when the case is not handled: be0c1a6e9ed03d43a10016330224c6c26c73b5e6